### PR TITLE
restrict m3 profile to admins only

### PR DESCRIPTION
**Pull request by ShanaLMoore**
@geekscruff Having a bit of trouble being able to easily utilize the user factory to create an admin in my specs. I have a workaround but feel that I shouldn't have to do it this way: https://notch8.slack.com/archives/CKY104K46/p1568418527000400

### CODE REVIEW

File: spec/models/ability_spec.rb [hunk 1]
```diff
@@ -53,4 +53,23 @@
...
+    before { ability = Ability.new(user) }
+
+    context 'when not admin' do
+      it "should not be manageable by a user of the buyer" do
+        expect(ability.can?(:manage, m3_profile)).to be_falsey
+      end
+    end
+
+    context 'when admin' do
+      it "should be manageable by a user of the buyer" do
```
Comment (ghost) [inline spec/models/ability_spec.rb]: tidy up the it description

File: spec/models/ability_spec.rb [hunk 2]
```diff
@@ -53,4 +53,23 @@
...
     end
   end
+
+  describe '#m3_profile_abilities' do
+    let(:m3_profile) { FactoryBot.create :m3_profile }
+
+    before { ability = Ability.new(user) }
+
+    context 'when not admin' do
+      it "should not be manageable by a user of the buyer" do
```
Comment (ghost) [inline spec/models/ability_spec.rb]: tidy up the it description

File: spec/models/ability_spec.rb [hunk 3]
```diff
@@ -53,4 +53,23 @@
       end
     end
   end
+
+  describe '#m3_profile_abilities' do
+    let(:m3_profile) { FactoryBot.create :m3_profile }
+
+    before { ability = Ability.new(user) }
```
Comment (ghost) [inline spec/models/ability_spec.rb]: I don't think this is needed, it will use the default from the top

File: spec/models/ability_spec.rb [hunk 4]
```diff
@@ -53,4 +53,23 @@
...
+
+    context 'when not admin' do
+      it "should not be manageable by a user of the buyer" do
+        expect(ability.can?(:manage, m3_profile)).to be_falsey
+      end
+    end
+
+    context 'when admin' do
+      it "should be manageable by a user of the buyer" do
+        user.roles << Role.where(name: 'admin').first_or_create
```
Comment (ghost) [inline spec/models/ability_spec.rb]: I couldn't figure out a way round this, would it be neater to put this line into a before block? (`before { user.roles << Role.where(name: 'admin').first_or_create }`)

**Review by ghost — CHANGES_REQUESTED** [review]

File: spec/models/ability_spec.rb (hunk 2 above)
Comment (ShanaLMoore) [reply spec/models/ability_spec.rb]: oops, good catch. Apparently my mind is still over in Tangibol!

File: spec/models/ability_spec.rb (hunk 3 above)
Comment (ShanaLMoore) [reply spec/models/ability_spec.rb]: yep, I hadn't noticed that. Thanks!

File: spec/models/ability_spec.rb (hunk 1 above)
Comment (ShanaLMoore) [reply spec/models/ability_spec.rb]: 👍

**Review by ghost — APPROVED** [review]

**Review by ghost — COMMENTED** [review]
Nice!

**Pull request merged**